frontend/src/components/taxons/EditTaxon.tsx

```diff
@@ -0,0 +1,109 @@
+import { Button, Form, Input, Space } from "antd";
+import React, { useCallback, useState } from "react";
+import { requiredRules } from "../../constants";
+import { useAppDispatch, useAppSelector } from "../../hooks";
+import { selectAppInitialzed, selectTaxonsByID } from "../../selectors";
+import { useNavigate, useParams } from "react-router-dom";
+import { add, update } from "../../modules/taxons/actions";
+import AppPageHeader from "../AppPageHeader";
+import PageContent from "../PageContent";
+
+export interface Taxon {
+    id?: number | null
+    ncbi_id: string
+    name: string
+}
+export interface EditTaxonProps {
+    taxon: Taxon
+}
+
+export const AddTaxonRoute = () => {
+    const emptyTaxon = { ncbi_id: "", name: "" }
+    const appInitialzed = useAppSelector(selectAppInitialzed)
+
+    return appInitialzed ? <EditTaxon taxon={emptyTaxon} /> : null
+
+}
+export const EditTaxonRoute = () => {
+    const taxons = useAppSelector(selectTaxonsByID)
+    const { id } = useParams();
+    const appInitialzed = useAppSelector(selectAppInitialzed)
+
+    return appInitialzed ? <EditTaxon taxon={taxons[id ?? -1]} /> : null
```

Comment:
I'm not sure what `taxons[-1]` is going to result in, but we should definitely not be creating the EditTaxon component if the id is undefined, nor if the taxon with the given id cannot be found. In those cases, the route should return null. The point of the route is to avoid ever passing an undefined Taxon prop to EditTaxon, so that we can write the EditTaxon code without having to check for an undefined `taxon` value every time we use it.

It's also a good idea to pass a clone of the taxon object as a prop, just to be sure the original copy doesn't get modified by accident.